ddd-koans/Entity/Lifecycle/src/Task.kt

```diff
@@ -20,10 +28,21 @@ class Bar(
     override fun toString(): String {
         return "Bar(ordinalNumber=$ordinal, notes=$notes, timeSignature=$timeSignature)"
     }
+
+    private fun validateState() {
+        state()
+    }
+
+    enum class State {
```

Comment:
so are we going to introduce `State` pattern? I guess in the reference book we use, it's mentioned as anti-pattern proposing having just explicit implementation of in our example, `IcompleteBar : Bar` `CompleteBar: Bar` which allows to expose behaviour only valid in given state like `not giving an opportunity to add more notes in CompleteBar` -> otherwise, with `State` pattern we need to match on each behaviour function what is allowed...just to consider

Reply:
of course, we can show evolution of that approach